### PR TITLE
Add support for config files.

### PR DESCRIPTION
**Pull request by kisom**

### CODE REVIEW

**Comment by kisom** [comment]
Pushed a change to bring tests for the new package up to 100%.

File: config/config_test.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+
+	}
+
+	empty.Merge(testConfig)
+	if !empty.equal(testConfig) {
+		t.Fatal("merging should result in equivalent configs")
+	}
+}
+
+// TestMergeOverride verifies that merges will keep the
```
Comment (lziest) [inline config/config_test.go]: broken sentence :)

**Comment by lziest** [comment]
LGTM

**Comment by lziest** [comment]
thanks for the backward compatibility.

File: config/config_test.go (hunk 1 above)
Comment (kisom) [inline config/config_test.go]: Fixed.

**Pull request merged**